src/yggdrasil/multicast.go

```diff
@@ -146,38 +136,55 @@ func (m *multicast) announce() {
 				if !addrIP.IsLinkLocalUnicast() {
 					continue
 				}
-				// Join the multicast group
-				m.sock.JoinGroup(&iface, groupAddr)
-				// Try and see if we already have a TCP listener for this interface
-				var listener *tcpListener
-				if l, ok := m.listeners[iface.Name]; !ok || l.listener == nil {
-					// No listener was found - let's create one
-					listenaddr := fmt.Sprintf("[%s%%%s]:%d", addrIP, iface.Name, m.listenPort)
-					if li, err := m.core.link.tcp.listen(listenaddr); err == nil {
-						m.core.log.Debugln("Started multicasting on", iface.Name)
-						// Store the listener so that we can stop it later if needed
-						m.listeners[iface.Name] = li
-						listener = li
-					} else {
-						m.core.log.Warnln("Not multicasting on", iface.Name, "due to error:", err)
-					}
-				} else {
-					// An existing listener was found
-					listener = m.listeners[iface.Name]
+				ipAddr := net.IPAddr{
+					IP:   addrIP,
+					Zone: iface.Name,
 				}
-				// Make sure nothing above failed for some reason
-				if listener == nil {
-					continue
-				}
-				// Get the listener details and construct the multicast beacon
-				lladdr := listener.listener.Addr().String()
-				if a, err := net.ResolveTCPAddr("tcp6", lladdr); err == nil {
-					a.Zone = ""
-					destAddr.Zone = iface.Name
-					msg := []byte(a.String())
-					m.sock.WriteTo(msg, nil, destAddr)
+				addresses[ipAddr.String()] = ipAddr
```

Comment:
I'm pretty sure that adding a `break` after this line would do the right thing in the case where there are multiple link-local addresses per link, *if* `iface.Addrs()` returns a slice that's sorted from oldest to newest. If not, then we could sort it ourselves, but weird things might happen if we add a new earlier address while keeping the old later one (listener closes, but connections don't, and then we could get a duplicate connection with the new listener).